### DISCUSSION TITLE
curl_global_sslset(-1, NULL, &avail) returns CURLSSLSET_TOO_LATE if MultiSSL not enabled?

### DISCUSSION THREAD

**Discussion by nojb** [curl/curl#7298] [Q&A] [answered]
Hello,

Is it expected for `curl_global_sslset(-1, NULL, &avail)` to return `CURLSSLSET_TOO_LATE` for `libcurl` builds **without** MultiSSL support? This is not clear from the documentation, but it seems to match what I am getting...

This is using the system `libcurl` under Ubuntu 18.04.5 LTS (bionic):
```
curl 7.58.0 (x86_64-pc-linux-gnu) libcurl/7.58.0 OpenSSL/1.1.1 zlib/1.2.11 libidn2/2.0.4 libpsl/0.19.1 (+libidn2/2.0.4) nghttp2/1.30.0 librtmp/2.3
Release-Date: 2018-01-24
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
```

Thanks!

**Comment by jay** [comment, marked as answer]
> Is it expected for `curl_global_sslset(-1, NULL, &avail)` to return `CURLSSLSET_TOO_LATE` for `libcurl` builds **without** MultiSSL support?

No it shouldn't do that. I cannot reproduce in curl-7_58_0 or master ca60a43 without multi. Please refer to [curl_global_sslset](https://curl.se/libcurl/c/curl_global_sslset.html) to see if you're using it properly, and also confirm the libcurl you are using is actually the same one as the curl tool that you showed above:

    curl_version_info_data *info = curl_version_info(CURLVERSION_NOW);
    printf("MultiSSL is %s\n", ((info->features & CURL_VERSION_MULTI_SSL) ? "on" : "off"));

If both of those things check out then maybe it is a bug that was fixed after 7.58.0, since the code is different:

https://github.com/curl/curl/blob/curl-7_58_0/lib/vtls/vtls.c#L1298-L1317
https://github.com/curl/curl/blob/curl-7_77_0/lib/vtls/vtls.c#L1389-L1416

**Reply by jay** [reply]
I can reproduce this with 7.58.0. I had rebuilt a different libcurl by mistake. So it is likely a bug that was fixed.

**Reply by nojb** [reply]
Thanks for the confirmation, I will try with a newer version.
